### ISSUE TITLE
Document project structure and general control flow

### ISSUE THREAD

**Issue by wxb1ank** [BadRobots1014/BadRobot2022#7]
Our *code* is fairly well-documented at the moment, but the file structure and 'bigger picture' ideas are not; I think it would be beneficial both for ourselves in the present and the programming team in the future to document these aspects. It might look like the following:

---

#### *commands/*

Autonomous and teleop. command implementations.

#### *commands/drive/*

'Drive strategy' implementations. These are not *quite* commands themselves; see `DriveStrategyCommand`.

#### *prototype/*

(...and so on.)

### Control Flow

1. [`Main::main`](https://github.com/BadRobots1014/BadRobot2022/blob/master/src/main/java/frc/robot/Main.java)
2. [`Robot::new`](https://github.com/BadRobots1014/BadRobot2022/blob/master/src/main/java/frc/robot/Robot.java)
3. [`Robot::robotInit`](https://github.com/BadRobots1014/BadRobot2022/blob/master/src/main/java/frc/robot/Robot.java)
4. [`RobotContainer::new`](https://github.com/BadRobots1014/BadRobot2022/blob/master/src/main/java/frc/robot/RobotContainer.java)

#### When <kbd>(shoot button)</kbd> is Pressed

1. The gatherer motor engages to collect cargo.
2. (...and so on.)

(...and so on. I'd either list major methods or describe the control flow in prose. You might also consider using LaTeX to generate a directed graph.)

---

While our file structure probably follows WPILib conventions, and the control flow can probably be surmised from reading enough WPILib documentation, it would be very convenient to centralize this information in our actual codebase.

**Comment by vchen04** [comment]
A convenient place to summarize the big-picture structure of the codebase is in a README.md at the root of the repository or on a repository wiki page.

**Comment by wxb1ank** [comment]
I agree; either of those would be great places for documentation/project overview. Maybe we could start with a README and flesh it out into a wiki if necessary.
